pkg/util/cloudprovider.go

```diff
@@ -48,3 +55,25 @@ func DetectCloudProvider() {
 	}
 	log.Info("No cloud provider detected")
 }
+
+// GetCloudProviderNTPHosts detects the cloud provider where the agent is running in order and returns its NTP host name.
+func GetCloudProviderNTPHosts() []string {
+	detectors := []cloudProviderNTPDetector{
+		{name: ecscommon.CloudProviderName, callback: ecs.GetNTPHosts},
+		{name: ec2.CloudProviderName, callback: ec2.GetNTPHosts},
+		{name: gce.CloudProviderName, callback: gce.GetNTPHosts},
+		{name: azure.CloudProviderName, callback: azure.GetNTPHosts},
+		{name: alibaba.CloudProviderName, callback: alibaba.GetNTPHosts},
+		{name: tencent.CloudProviderName, callback: tencent.GetNTPHosts},
+	}
+
+	for _, cloudNTPdetector := range detectors {
+		if cloudNTPserver := cloudNTPdetector.callback(); cloudNTPserver != nil {
+			log.Debug("Cloud provider %s detected", cloudNTPdetector.name)
```

Comment:
We should log something about NTP here in the wording:

What about ?
```suggestion
			log.Debug("Using NTP servers from %s cloud provider", cloudNTPdetector.name)
```